addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -273,23 +287,20 @@ export class ProductScreen extends Component {
         }
     }
     async _getProductByBarcode(code) {
-        let product = this.pos.db.get_product_by_barcode(code.base_code);
+        let product = this.pos.models["product.product"].getBy("barcode", code.base_code);
+
         if (!product) {
-            // find the barcode in the backend
-            const { product_id = [], packaging = [] } = await this.orm.silent.call(
+            const records = await this.pos.data.callRelated(
                 "pos.session",
                 "find_product_by_barcode",
                 [odoo.pos_session_id, code.base_code]
             );
-            if (product_id.length) {
-                await this.pos._addProducts(product_id, false);
-                if (packaging.length) {
-                    this.pos.db.add_packagings(packaging);
-                }
-                // assume that the result is unique.
-                product = this.pos.db.get_product_by_id(product_id[0]);
```

Comment:
What happened to the `product.packaging` records? It seems that it's not being taken into account anymore.

Reply:
They are automatically loaded by data_service